web/lib/api/ws.ts

```diff
@@ -0,0 +1,138 @@
+import {
+  ClientMessage,
+  ClientMessageType,
+  ServerMessage,
+} from 'common/api/websockets'
+import { getApiUrl } from 'common/api/utils'
+
+// mqp: useful for debugging
+const VERBOSE_LOGGING = true
+
+export type BroadcastHandler = (msg: ServerMessage<'broadcast'>) => void
+
+export class APIRealtimeClient {
+  ws!: WebSocket
```

Comment:
Wow, I guess WebSocket is a browser-defined type.

Reply:
Browser web sockets are like 10+ years old. The browser WebSocket client implementation is also provided with Node in the newest Node (22), but there's no WebSocket server in Node, so we will still need to use a library for that for the foreseeable future.